org.civicrm.reqangular/src/common/services/pub-sub-service.js

```diff
@@ -0,0 +1,32 @@
+define([
+  'common/modules/services'
+], function (module) {
+  'use strict';
+
+  var topics = Object.create(null);
+  module.factory("pubSubService", ['$rootScope', function ($rootScope) {
+    return {
+      subscribe: function (topic, listener) {
+        if (!topics[topic]) {
+          topics[topic] = [];
+        }
+        listener.$rootScope = $rootScope;
+        var index = topics[topic].push(listener) - 1;
+        return {
+          remove: function () {
+            delete topics[topic][index];
+          }
+        };
+      },
+      publish: function (topic, info) {
+        if (!topics[topic]) return;
+        topics[topic].forEach(function (item) {
+          item(info != undefined ? info : {});
```

Comment:
Generally always an object is sent and in the subscribe methods we check `data.property`, so incase of not sending any value, those cases will generate error. This is to avoid that.